core/app/models/concerns/spree/metafields.rb

```
@@ -0,0 +1,49 @@
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :metafields, -> { includes(:metafield_definition) }, as: :resource, class_name: 'Spree::Metafield', dependent: :destroy
+      has_many :public_metafields, -> { includes(:metafield_definition).available_on_front_end }, as: :resource, class_name: 'Spree::Metafield'
+      has_many :private_metafields, -> { includes(:metafield_definition).available_on_back_end }, as: :resource, class_name: 'Spree::Metafield'
+
+      accepts_nested_attributes_for :metafields, allow_destroy: true, reject_if: lambda { |mf|
+                                                                                     mf[:metafield_definition_id].blank? || (mf[:id].blank? && mf[:value].blank?)
+                                                                                   }
+
+      scope :with_metafield_key, ->(namespace, key) { joins(:metafield_definitions).where(spree_metafield_definitions: { namespace: namespace, key: key }) }
```

RuboCop (1.80.2)

[convention] Line is too long. [156/150]

(Layout/LineLength)